src/check_addresses_and_amounts.c

```diff
@@ -0,0 +1,257 @@
+    } else if (G_swap_ctx.subcommand == SELL || G_swap_ctx.subcommand == SELL_NG) {
+        amount = (pb_bytes_array_16_t *) &G_swap_ctx.sell_transaction.in_amount;
+        dest = G_swap_ctx.printable_send_amount;
+        dest_size = sizeof(G_swap_ctx.printable_send_amount);
+    } else {
+        amount = (pb_bytes_array_16_t *) &G_swap_ctx.fund_transaction.in_amount;
+        dest = G_swap_ctx.printable_send_amount;
+        dest_size = sizeof(G_swap_ctx.printable_send_amount);
+    }
+    if (get_printable_amount(&sub_coin_config,
+                             appname,
+                             amount->bytes,
+                             amount->size,
+                             dest,
+                             dest_size,
+                             false) < 0) {
+        PRINTF("Error: Failed to get printable amount\n");
+        return false;
+    }
+    PRINTF("Formatted amount: %s\n", dest);
+    return true;
+}
+
+static bool format_fees(buf_t sub_coin_config, char *appname) {
+    if (get_printable_amount(&sub_coin_config,
+                             appname,
+                             G_swap_ctx.transaction_fee,
+                             G_swap_ctx.transaction_fee_length,
+                             G_swap_ctx.printable_fees_amount,
+                             sizeof(G_swap_ctx.printable_fees_amount),
+                             true) < 0) {
+        PRINTF("Error: Failed to get printable fees amount\n");
+        return false;
+    }
+    PRINTF("Fees: %s\n", G_swap_ctx.printable_fees_amount);
+    return true;
+}
+
+static bool format_fiat_amount(void) {
+    size_t len = strlen(G_swap_ctx.sell_transaction.out_currency);
+    if (len + 1 >= sizeof(G_swap_ctx.printable_get_amount)) {
+        return false;
+    }
+
+    strncpy(G_swap_ctx.printable_get_amount,
+            G_swap_ctx.sell_transaction.out_currency,
+            sizeof(G_swap_ctx.printable_get_amount));
+    G_swap_ctx.printable_get_amount[len] = ' ';
+    G_swap_ctx.printable_get_amount[len + 1] = '\x00';
+
+    if (get_fiat_printable_amount(G_swap_ctx.sell_transaction.out_amount.coefficient.bytes,
+                                  G_swap_ctx.sell_transaction.out_amount.coefficient.size,
+                                  G_swap_ctx.sell_transaction.out_amount.exponent,
+                                  G_swap_ctx.printable_get_amount + len + 1,
+                                  sizeof(G_swap_ctx.printable_get_amount) - (len + 1)) < 0) {
+        PRINTF("Error: Failed to get source currency printable amount\n");
+        return false;
+    }
+
+    PRINTF("%s\n", G_swap_ctx.printable_get_amount);
+    return true;
+}
+
+static void format_account_name(void) {
+    strncpy(G_swap_ctx.account_name,
+            G_swap_ctx.fund_transaction.account_name,
+            sizeof(G_swap_ctx.account_name));
+    G_swap_ctx.account_name[sizeof(G_swap_ctx.account_name) - 1] = '\x00';
+}
+
+int check_addresses_and_amounts(const command_t *cmd) {
```

Comment:
If your ambition in the future is to deprecate current `SWAP`, `FUND` and `SELL` in favor of `NG` transactions, you may have an interest into keeping two entrypoint functions here, one for the legacy transactions, one the ng ones.

Reply:
Mmmh not sure, as it is I just have to remove the ledgacy conditions from the `if` and it will work out of the box

Reply:
At this point of the transaction there is no handling of signature types / encoding anymore, so it's 100% transparent to the code if we are in legacy or unified